cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -185,8 +188,10 @@ public BodyConsumer create(HttpRequest request, HttpResponder responder,
                              @PathParam("namespace-id") final String namespaceId,
                              @PathParam("app-id") final String appId)
     throws BadRequestException, NamespaceNotFoundException, AccessException {
-
-    ApplicationId applicationId = validateApplicationId(namespaceId, appId);
+    RunId runId = RunIds.generate();
+    ApplicationId applicationId = validateApplicationVersionId(namespaceId, appId,
+                                                               RunIds.getTime(runId, TimeUnit.MILLISECONDS)
```

Comment:
Why we can't just the UUID as the version? Why we want to prefix it with the timestamp, which is actually already encoded inside the UUID?

Reply:
discussed offline - will link the JIRA

Reply:
@sanjanasandeep we could remove the time prefix in this PR. Will fix this in https://cdap.atlassian.net/browse/CDAP-19781

Reply:
got it - will GET /apps work fine after removing it tho?

Reply:
the version history will show versions not in creation time order, it will be sorted on the uuid